### PR TITLE
Upgrade spotinst/kubernetes-cluster-controller from 1.0.39 to 1.0.47

### PR DESCRIPTION
**Pull request by povils**
Spotinst has kubernetes controller of version 1.0.47: https://api.spotinst.com/ocean/tutorials/spotinst-kubernetes-controller/changelog/

### CODE REVIEW

**Comment by rifelpet** [comment]
Looking at the installation docs using kubectl I see there are some other differences in the manifest:

https://api.spotinst.com/ocean/tutorials/spotinst-kubernetes-controller/installing-kubernetes-controller/installing-kubernetes-controller-kubectl/

I see new env vars being set as well as RBAC updates, priority class, node affinity, and a few other minor changes. In case the env vars are required in the new version it might be a good idea to update the rest of the Kops manifest to match what spotinst recommends.

**Comment by povils** [comment]
@rifelpet Thanks for noticing!

File: upup/models/cloudup/resources/addons/spotinst-kubernetes-cluster-controller.addons.k8s.io/v1.8.0.yaml.template [hunk 1]
```diff
@@ -84,33 +97,47 @@ apiVersion: apps/v1beta1
...
+    k8s-app: spotinst-kubernetes-cluster-controller
   name: spotinst-kubernetes-cluster-controller
   namespace: kube-system
 spec:
   replicas: 1
   revisionHistoryLimit: 10
   selector:
     matchLabels:
-      k8s-addon: spotinst-kubernetes-cluster-controller.addons.k8s.io
+      k8s-app: spotinst-kubernetes-cluster-controller
```
Comment (rifelpet) [inline upup/models/cloudup/resources/addons/spotinst-kubernetes-cluster-controller.addons.k8s.io/v1.8.0.yaml.template]: Unfortunately this field isnt mutable, so updating the Deployment resource will fail. I think we should keep the `k8s-addon` label on all the resources but update everything else to match the upstream manifest :( it is not ideal.

**Comment by rifelpet** [comment]
This was upgraded in https://github.com/kubernetes/kops/pull/8080 so I think this PR is safe to close. We appreciate the contribution though!
/close

**Pull request closed without merging**